plugins/importexport/onix30/filter/MonographONIX30XmlFilter.inc.php

```diff
@@ -322,20 +323,20 @@ function createProductNode($doc, $submission, $publicationFormat) {
 
 		$productTitleDetailNode->appendChild($titleElementNode);
 
-		if (!$publication->getLocalizedData('prefix') || !$publication->getLocalizedData('title')) {
-			$titleElementNode->appendChild($this->_buildTextNode($doc, 'TitleText', trim($publication->getLocalizedData('prefix') ?? $publication->getLocalizedTitle())));
+		if (!$publication->getLocalizedData('prefix', $pubLocale) || !$publication->getLocalizedData('title', $pubLocale)) {
+			$titleElementNode->appendChild($this->_buildTextNode($doc, 'TitleText', trim($publication->getLocalizedData('prefix', $pubLocale) ?? $publication->getLocalizedTitle($pubLocale))));
 		} else {
-			if ($publication->getLocalizedData('prefix')) {
-				$titleElementNode->appendChild($this->_buildTextNode($doc, 'TitlePrefix', $publication->getLocalizedData('prefix')));
+			if ($publication->getLocalizedData('prefix', $pubLocale)) {
+				$titleElementNode->appendChild($this->_buildTextNode($doc, 'TitlePrefix', $publication->getLocalizedData('prefix', $pubLocale)));
 			} else {
 				$titleElementNode->appendChild($doc->createElementNS($deployment->getNamespace(), 'NoPrefix'));
 			}
 
-			$titleElementNode->appendChild($this->_buildTextNode($doc, 'TitleWithoutPrefix', $publication->getLocalizedTitle()));
+			$titleElementNode->appendChild($this->_buildTextNode($doc, 'TitleWithoutPrefix', $publication->getLocalizedTitle($pubLocale)));
```

Comment:
I think that also here we should use getData -- the element seems to want only title without prefix and getLocalizedTitle would return the prefix as well, if it exists.